### PR TITLE
Add -additionalplugins flag

### PR DESCRIPTION
**Pull request by sowelipililimute**
This makes the orchestration of multiple similar-but-not-identical TShock instances easier; as shared plugins can go in the default ServerPlugins directory, and instance-specific plugins can go in a directory passed via this flag when launching an instance.

### CODE REVIEW

**Review by hakusaro — APPROVED** [review]

**Comment by hakusaro** [comment]
This seems reasonable!

File: TerrariaServerAPI/TerrariaApi.Server/ServerApi.cs
Comment (Arthri) [inline TerrariaServerAPI/TerrariaApi.Server/ServerApi.cs:313]:
```suggestion
			if (AdditionalPluginsPath is not null)
```
imo, using a null check could express its purpose better. Reference: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0150

**Review by Arthri — CHANGES_REQUESTED** [review]

File: TerrariaServerAPI/TerrariaApi.Server/ServerApi.cs
Comment (hakusaro) [reply TerrariaServerAPI/TerrariaApi.Server/ServerApi.cs:313]: Applying this suggestion would break L315, would it not?

Comment (SignatureBeef) [reply TerrariaServerAPI/TerrariaApi.Server/ServerApi.cs:313]: > Applying this suggestion would break L315, would it not?

yes it would. the original code looks fine to me. if we actually turn on nullable contexts, the code here already tests for nulls. if it gets changed though, L315 would need to be altered to the null checked variable. i personally wouldn't stress about an extra variable, but thats me.

Comment (sowelipililimute) [reply TerrariaServerAPI/TerrariaApi.Server/ServerApi.cs:313]: so, is this fine as-is or nah?

Comment (hakusaro) [reply TerrariaServerAPI/TerrariaApi.Server/ServerApi.cs:313]: it's fine to me; external PRs require to approval buttons to merge though, so we're just waiting on one more approval

Comment (Arthri) [reply TerrariaServerAPI/TerrariaApi.Server/ServerApi.cs:313]: I don't mind. LGTM

**Review by Arthri — APPROVED** [review]

**Pull request merged**
